src/utils/web3.js

```diff
@@ -79,10 +79,19 @@ const getERC20Parameters = async (smartContract) => {
   };
 };
 
+const mapSmartContractToDictionary = (abi) => {
+  const result = {};
+  abi.forEach(item => result[item.name] = item.outputs && item.outputs.length > 0 ? item.outputs[0].type : null);
+  return result;
+};
+
 export const getICOParameters = async (web3, address) => {
   // tokenContract may be different that ICO contract that governs ICO process
   const tokenContractAddress = config.ICOs[address].tokenContract || address;
   const tokenContract = getSmartContract(web3, tokenContractAddress);
+
+  const smartContractDic = mapSmartContractToDictionary(tokenContract.abi);
```

Comment:
please do not write code that is doing two unrelated things. getICOParameters has nothing to do with providing ABI. I think better way to handle this would be to make a method that returns tokenSmartContract then get ABI from it and then use it to call getICOParameters (instead of address). OK?